[ASM] Single Step ASM Helm chart (#23182)
DataDog/documentation: content/en/security/application_security/enabling/single_step/code_security.md

## Requirements

- **Minimum Agent version 7.53.0**

> - **Minimum Helm version 3.62.0** (For Kubernetes deployments)

Author: ```suggestion
- **Minimum Datadog Helm chart version 3.62.0** (For Kubernetes deployments)
```

[6]: /service_catalog/
[7]: https://github.com/DataDog/dd-trace-js?tab=readme-ov-file#version-release-lines-and-maintenance

{{% /tab %}}

{{% tab "Kubernetes" %}}

You can enable APM by installing the Agent with the Datadog Helm chart. This deploys the Datadog Agent across all nodes in your Linux-based Kubernetes cluster with a DaemonSet.

> **Note**: Single step instrumentation doesn't instrument applications in the namespace where you install the Datadog Agent. It's recommended to install the Agent in a separate namespace in your cluster where you don't run your applications.

Reviewer: ```suggestion
<div class="alert alert-info">Single step instrumentation doesn't instrument applications in the namespace where you install the Datadog Agent. It's recommended to install the Agent in a separate namespace in your cluster where you don't run your applications.</div>
```

### Requirements

- Make sure you have [Helm][13] installed.

### Installation

To enable single step instrumentation with Helm:

1. Add the Helm Datadog repo:
   ```bash
    helm repo add datadog https://helm.datadoghq.com
    helm repo update
    ```

> 2. Create a Kubernetes secret to store your Datadog [API key][10]:

Reviewer: ```suggestion
2. Create a Kubernetes Secret to store your Datadog [API key][10]:
```

   ```bash
   kubectl create secret generic datadog-secret --from-literal api-key=$DD_API_KEY
   ```
3. Create `datadog-values.yaml` and add the following configuration:
   ```
   datadog:
    apiKeyExistingSecret: datadog-secret
    site: <DATADOG_SITE>
    apm:
      instrumentation:
         enabled: true
    asm:
      iast:
        enabled: true
   ```
   Replace `<DATADOG_SITE>` with your [Datadog site][12].

4. Run the following command:
   ```bash
   helm install datadog-agent -f datadog-values.yaml datadog/datadog
   ```
5. Do a rolling restart on your applications for instrumentation to take effect.

> More information on Kubernetes Single Step Instrumentation here:

Reviewer: ```suggestion
For more information on Kubernetes single step instrumentation, see the following:
```

3. To disable ASM for code security, remove the `DD_IAST_ENABLED=true` environment variable from your application configuration, and restart your service.
{{% /tab %}}

{{% tab "Kubernetes" %}}

> 1. Set the `admission.datadoghq.com/enabled:` label to `"false"` for the pod spec:
>    ```yaml
>    spec:
>      template:
>        metadata:
>          labels:
>            admission.datadoghq.com/enabled: "false"
>    ```
> 2. Apply the configuration:
>    ```bash
>    kubectl apply -f /path/to/your/deployment.yaml
>    ```
> 3. Restart the services you want to remove instrumentation for.

Reviewer: ```suggestion
1. Set the `admission.datadoghq.com/enabled:` label to `"false"` for the pod spec:
  
   ```yaml
   spec:
     template:
       metadata:
         labels:
           admission.datadoghq.com/enabled: "false"
   ```
2. Apply the configuration:
  
   ```bash
   kubectl apply -f /path/to/your/deployment.yaml
   ```
3. Restart the services you want to remove instrumentation for.
```

> **Note:** You can disable ASM code security while keeping APM up by adding the `DD_IAST_ENABLED=false` environment variable to your deployments.

Reviewer: ```suggestion
<div class="alert alert-info"> You can disable ASM code security while keeping APM up by adding the <code>DD_IAST_ENABLED=false</code> environment variable to your deployments.</div>
```

{{% /tab %}}

{{% tab "Kubernetes" %}}

1. Under `apm:`, remove `instrumentation:` and all following configuration in `datadog-values.yaml`.
2. Under `asm:`, remove `iast:` and all following configuration in`datadog-values.yaml`.

> 3. Run the following command:
>    ```bash
>    helm upgrade datadog-agent -f datadog-values.yaml datadog/datadog
>    ```
> 4. Restart your services.

Reviewer: ```suggestion
3. Run the following command:
  
    ```bash
   helm upgrade datadog-agent -f datadog-values.yaml datadog/datadog
   ```
4. Restart your services.
```
